ledger/src/blockstore.rs

```diff
@@ -3327,8 +3327,15 @@ impl Blockstore {
     /// roots themselves. Then, mark these found slots as roots since the
     /// ancestor of a root is also inherently a root.
     pub fn scan_and_fix_roots(&self, exit: &AtomicBool) -> Result<()> {
-        let ancestor_iterator = AncestorIterator::new(self.last_root(), self)
-            .take_while(|&slot| slot >= self.lowest_cleanup_slot());
+        // Hold the lowest_cleanup_slot read lock to prevent any cleaning of
+        // the blockstore from another thread. Doing so will prevent a
+        // possible inconsistency across column families where a slot is:
+        //  - Identified as needing root repair by this thread
+        //  - Cleaned from the blockstore by another thread (LedgerCleanupSerivce)
+        //  - Marked as root via Self::set_root() by this this thread
+        let lowest_cleanup_slot = self.lowest_cleanup_slot.read().unwrap();
```

Comment:
I'm curious whether you saw this cause an issue somewhere? I doubt it comes up a lot, since we fixed the issue that was causing regular missing roots (I'm not sure how many RPC nodes are even running with `rpc_scan_and_fix_roots` these days), but definitely good to tighten up. Thanks for fixing!